tubul/tubul_logger.h

```diff
@@ -0,0 +1,80 @@
+//
+// Created by Nicolas Loira on 2/3/23.
+//
+
+#pragma once
+
+#include <cstdint>
+#include "tubul_string.h"
+// #include "tubul_engine.h"
+
+namespace TU {
+    enum class LogLevel : uint8_t {
+        ERROR,
+        WARNING,
+        REPORT,
+        INFO,
+        DEVEL,
+        STATS,
+        DEBUG
+    };
+
+    enum class LogOptions : uint8_t {
+        NONE = 0,
+        COLOR = 1,          // send commands for color output
+        EXCLUSIVE = 2,      // only send to specified LogLevel
+        NOTIMESTAMP = 4,    // don't prefix timestamp
+        QUIET = 8           // don't show anything
+    };
+
+    inline LogOptions operator|(LogOptions a, LogOptions b) {
+        return static_cast<LogOptions>(static_cast<uint8_t>(a) | static_cast<uint8_t>(b));
```

Comment:
Estos casts raros los podemos mejorar usando std::underlying_type. Crearía una funcion tipo TU::asInt para que haga el trabajo sucio y retorne un int o algo asi. Queda un poco mas agradable a la vista.

Reply:
Me gusta la idea del `std::underlying_type`. No sé si usaría el `TU::asInt`, porque al final tenemos que retornar un `LogOptions` y tendríamos que volver a explicitar el "underlying type" en la función.

Es decir, lo dejaría como:
```
inline LogOptions operator|(LogOptions a, LogOptions b) {
        return static_cast<LogOptions>(static_cast<std::underlying_type(LogOptions)>(a) | static_cast<std::underlying_type(LogOptions)>(b));
```

(bien feíto, la verdad)